[Operational Issue]: Flaky file lock issue in Microsoft.AspNetCore.Components.Gateway

### Error

> ##[error].dotnet\sdk\11.0.100-preview.6.26359.118\Sdks\Microsoft.NET.Sdk.StaticWebAssets\targets\Microsoft.NET.Sdk.StaticWebAssets.Compression.targets(359,5): error : The process cannot access the file 'D:\a_work\1\s\src\aspnetcore\artifacts\obj\Microsoft.AspNetCore.Components.Gateway\Release\net11.0\compressed\publish\o4b6igg9d4-{0}-urybgy2z5y-urybgy2z5y.br' because it is being used by another process.

### Links

- build log (attempt 1): https://dev.azure.com/dnceng/internal/_build/results?buildId=3030850&view=logs&j=e45a3293-4024-5a66-d70f-cda470bd6fb5&s=56cc5bc2-3154-5d19-920f-2db03d3b8537&t=4bb57555-d7dc-5e8e-b80e-1a898a2fd87a&l=2952)

### Root Cause

unknown

### Blocking

- [ ] This issue is blocking

### Affected Releases

- [ ] .NET 10
- [ ] .NET 10 (SDK band build)
- [x] .NET 11

--- a/src/Components/Gateway/cli/Microsoft.AspNetCore.Components.Gateway.Cli.csproj
+++ b/src/Components/Gateway/cli/Microsoft.AspNetCore.Components.Gateway.Cli.csproj
@@ -37,11 +37,19 @@
                       PrivateAssets="All" />
   </ItemGroup>
 
-  <!-- Build and publish the gateway before packing so its binaries are available to the nuspec. -->
+  <!-- Build and publish the gateway before packing so its binaries are available to the nuspec.
+       Gateway.csproj also ships its own package, so it's already built directly as a top-level
+       project elsewhere in the overall (parallel) build. Forcing an explicit TargetFramework here
+       gave this MSBuild request a global property that the other invocation doesn't have (Gateway.csproj
+       is single-targeted, so it doesn't need one), making MSBuild treat the two as distinct, unrelated
+       build instances. Both ended up running concurrently and racing to write the same compressed
+       static web asset files (see https://github.com/dotnet/dotnet/issues/8005). Not overriding
+       Configuration/TargetFramework here lets this request's global properties match the other
+       invocation's, so MSBuild recognizes it as the same request and reuses/waits for its result
+       instead of duplicating it. -->
   <Target Name="_PublishGatewayForTool" BeforeTargets="GenerateNuspec">
     <MSBuild Projects="$(_GatewayProject)"
-             Targets="Publish"
-             Properties="Configuration=$(Configuration);TargetFramework=$(DefaultNetCoreTargetFramework)" />
+             Targets="Publish" />
 
     <Error Condition="!Exists('$(_GatewayPublishDir)blazor-gateway.dll')"
            Text="Expected the gateway to publish 'blazor-gateway.dll' to '$(_GatewayPublishDir)' but it was not found." />

--- a/src/Components/Gateway/cli/Microsoft.AspNetCore.Components.Gateway.Cli.csproj
+++ b/src/Components/Gateway/cli/Microsoft.AspNetCore.Components.Gateway.Cli.csproj
@@ -38,15 +38,9 @@
   </ItemGroup>
 
   <!-- Build and publish the gateway before packing so its binaries are available to the nuspec.
-       Gateway.csproj also ships its own package, so it's already built directly as a top-level
-       project elsewhere in the overall (parallel) build. Forcing an explicit TargetFramework here
-       gave this MSBuild request a global property that the other invocation doesn't have (Gateway.csproj
-       is single-targeted, so it doesn't need one), making MSBuild treat the two as distinct, unrelated
-       build instances. Both ended up running concurrently and racing to write the same compressed
-       static web asset files (see https://github.com/dotnet/dotnet/issues/8005). Not overriding
-       Configuration/TargetFramework here lets this request's global properties match the other
-       invocation's, so MSBuild recognizes it as the same request and reuses/waits for its result
-       instead of duplicating it. -->
+       No Properties override: Gateway.csproj is also built as a top-level project elsewhere in the
+       build, and forcing TargetFramework here made MSBuild treat the two as separate, racing builds
+       (see https://github.com/dotnet/dotnet/issues/8005). -->
   <Target Name="_PublishGatewayForTool" BeforeTargets="GenerateNuspec">
     <MSBuild Projects="$(_GatewayProject)"
              Targets="Publish" />

